pkg/ibmvpc/examples/out/analysis_out/sg_testing1_new_respond_partly_all_vpcs__with_grouping.txt

```diff
@@ -1,10 +1,10 @@
 Endpoint connectivity for VPC test-vpc1-ky
 Public Internet 147.235.219.206/32 => vsi2-ky[10.240.20.4] : protocol: TCP dst-ports: 22
 db-endpoint-gateway-ky[10.240.30.6],vsi3a-ky[10.240.30.5],vsi3b-ky[10.240.30.4] => db-endpoint-gateway-ky[10.240.30.6],vsi3a-ky[10.240.30.5] : All Connections
-db-endpoint-gateway-ky[10.240.30.6],vsi3a-ky[10.240.30.5],vsi3b-ky[10.240.30.4] => vsi1-ky[10.240.10.4] : protocol: ICMP,UDP; protocol: TCP src-ports: 100-200 dst-ports: 10-50
```

Comment:
Why this change? It makes the results far less readable

Reply:
it just replaces UDP with TCP, because of sorting

Reply:
Then I would expect it to be something like:
`protocol: ICMP; protocol: TCP src-ports: 100-200 dst-ports: 10-50; protocol: UDP`

Reply:
it could be like this if we do not have the special case that Elazar added line 119 strings file

Reply:
should I remove that special case? it just calls the tcpudpstring with short version

Reply:
actually we need it for other cases like `protocol: ICMP,UDP`
without the special case it will be `protocol: ICMP; protocol: UDP` and will chane a lot of files

Reply:
Suggesting to modify the `Stringify` code as follows:
1. Identify protocols which the given `TransportSet` covers fully
2. Identify cubes which are partial to the full protocol
3. Join results from 1. and 2. together.

Something like:
```golang
func Stringify(c *netset.TransportSet, shortVersion bool) string {
	if c.IsEmpty() {
		return NoConnections
	} else if c.IsAll() {
		return AllConnections
	}

	resStrs := []string{}
	resStrs = appendIfNotEmpty(resStrs, completeProtocols(c, shortVersion))
	resStrs = appendIfNotEmpty(resStrs, iCMPString(c.ICMPSet(), shortVersion)) // partial cubes only
	resStrs = appendIfNotEmpty(resStrs, tCPUDPString(c.TCPUDPSet(), shortVersion)) // partial cubes only

	return strings.Join(resStrs, semicolonString)
}
```

Where `completeProtocols()` is something like:
```golang
func completeProtocols(c *netset.TransportSet, shortVersion bool) string {
	completeProtocols := []string{}
	if netset.AllICMPSet().IsSubset(c.ICMPSet()) {
		completeProtocols = append(completeProtocols, string(netp.ProtocolStringICMP))
	}
	if netset.AllTCPTransport().IsSubset(c) {
		completeProtocols = append(completeProtocols, string(netp.ProtocolStringTCP))
	}
	if netset.AllUDPTransport().IsSubset(c) {
		completeProtocols = append(completeProtocols, string(netp.ProtocolStringUDP))
	}

	res := strings.Join(completeProtocols, ",")
	if !shortVersion && res != "" {
		res = protocolString + res
	}

	return res
}
```
I believe this will not require too many changes to test results